scala/private/common.bzl

```diff
@@ -87,33 +92,23 @@ def filter_not_sources(deps):
 
 def add_labels_of_jars_to(jars2labels, dependency, all_jars, direct_jars):
   for jar in direct_jars:
-    _add_label_of_direct_jar_to(jars2labels, dependency, jar)
+    jars2labels[jar.path] = dependency.label
   for jar in all_jars:
-    _add_label_of_indirect_jar_to(jars2labels, dependency, jar)
-
-def _add_label_of_direct_jar_to(jars2labels, dependency, jar):
-  jars2labels[jar.path] = dependency.label
-
-def _add_label_of_indirect_jar_to(jars2labels, dependency, jar):
-  if _label_already_exists(jars2labels, jar):
-    return
-
-  # skylark exposes only labels of direct dependencies.
-  # to get labels of indirect dependencies we collect them from the providers transitively
-  if _provider_of_dependency_contains_label_of(dependency, jar):
-    jars2labels[jar.path] = dependency.jars_to_labels[jar.path]
+    path = jar.path
+    if path not in jars2labels:
```

Comment:
this seems to introduce order dependency: if you find an unknown provider first, but later it is seen as a direct dependency, you'll only see the unknown, right?

Reply:
not for direct but if you find an unknown and then another sub graph finds it as direct and afterwards a third target depends on both of these then indeed it can miss the known label.
1. We can fix it by also checking if the path starts with `Unknown label` (I'd rather we not do it on this PR)
2. This should not happen now that we're moving everything to scala_import/scala_import_external
3. We're considering dropping this mechanism of passing the jars2labels in favor of what java_rules are doing (reading the labels of "offending" jars from the manifest and then using (offline) bazel query to understand which target exactly to add [for example support exports])